package-parser/package_parser/commands/generate_annotations/generate_annotations.py

```diff
@@ -273,9 +275,37 @@ def __add_implicit_usages_of_default_value(usages: UsageStore, api: API) -> None
             usages.add_value_usage(parameter_qname, default_value, location)
 
 
+def __get_optional_annotations(
+    usages: UsageStore, api: API, annotations: AnnotationStore
+) -> None:
+    """
+    Collects all parameters that are currently required but should be optional to be assign a value
+    :param usages: Usage store
+    :param api: Description of the API
+    :param annotations: AnnotationStore, that holds all annotations
+    """
+    parameters = api.parameters()
+    # Takes all parameters with default value
+    required_parameter = [(it, parameters[it]) for it in parameters]
```

Comment:
Comment & variable name does not match behavior. `required_parameter` contains all parameters rather than just required ones. This also means that `@Optional` annotations are created for parameters that are already optional.

1. Is this a deliberate decision?
2. Does it match the behavior of `@Required` annotations, i.e. are `@Required` annotations also created for parameters that are required anyway?